src/apps/backend/modules/account/errors.py

```diff
@@ -12,6 +12,24 @@ def __init__(self, message: str) -> None:
         super().__init__(code=AccountErrorCode.NOT_FOUND, http_status_code=404, message=message)
 
 
+class AccountWithUsernameNotFoundError(AccountNotFoundError):
+    def __init__(self, username: str) -> None:
+        message = f"Account with username:: {username}, not found"
```

Comment:
As I highlighted earlier, we need to offer solution, and write better error messages here and other places in this PR.

Whenever I give feedback, it is not only true for the site that was given but rather meant as a guidance to do a self review of the PR and make sure that feedback is followed across all the changes being proposed in the PR.